### ISSUE TITLE
Electron Cash source code doesn't run on various linux systems

### ISSUE THREAD

**Issue by richardamullens** [Electron-Cash/Electron-Cash#2052]
`I downloaded the source code tar package, extracted it, changed to its directory and then issued the command ./electron-cash with the following result (this is after seemingly succeeding to sudo pip3 install dns, pyaes, ecdsa, requests, qrcode and google)

richard@richard-XPS-13-9360 ~/Downloads $ cd Electron-Cash-4.2.0/
richard@richard-XPS-13-9360 ~/Downloads/Electron-Cash-4.2.0 $ python3 --version
Python 3.7.8
richard@richard-XPS-13-9360 ~/Downloads/Electron-Cash-4.2.0 $ ./electron-cash
Error: No module named 'google'. Try 'sudo pip install <module-name>'
richard@richard-XPS-13-9360 ~/Downloads/Electron-Cash-4.2.0 $ 

Prior to my attempts at fixing the missing resources it gave the error
Error: No module named 'dns'. Try 'sudo pip install <module-name>'

This error occurred on Linux Mint 18.3 and also on a newly installed Raspberry Pi 4 running Python 3.7.3

A 64 bit version of the raspberry pi operating system is still (after many months) in Beta and the AppImage requires 64 bits and so can't be run on a 32 bit OS.

I can see that electron-cash tests for the presence of various pure-python dependencies but the diagnostic messages it issues aren't clear enough for a naive user to rectify the problem`

**Comment by richardamullens** [comment]
My text above includes angle brackets but these and what lies between them aren't visible.
When it says Try sudo pip install it is missing "module name" in angle brackets.
Sorry, I'm unfamiliar with Git.

`Test`

**Comment by cculianu** [comment]
Do `python3 -m pip install --user google`, etc, to install the missing modules.

**Comment by cculianu** [comment]
You really should try using the source-dist, not the source tarball. [This from the "releases" page](https://github.com/Electron-Cash/Electron-Cash/releases/download/4.2.0/Electron-Cash-4.2.0.tar.gz)

**Comment by richardamullens** [comment]
Thank you for your responses.
The problem is that I had already installed "google" (possibly the wrong version) and when I started electron-cash, it complained that it wasn't there.  So

sudo python3 -m pip uninstall google
Uninstalling google-3.0.0:
  Would remove:
    /usr/local/bin/google
    /usr/local/lib/python3.7/dist-packages/google-3.0.0.dist-info/*
    /usr/local/lib/python3.7/dist-packages/googlesearch/*
Proceed (y/n)? 
Your response ('') was not one of the expected responses: y, n
Proceed (y/n)? y
  Successfully uninstalled google-3.0.0
pi@raspberrypi:~/Electron-Cash-4.2.0 $ python3 -m pip install --user google
Looking in indexes: https://pypi.org/simple, https://www.piwheels.org/simple
Collecting google
  Downloading https://files.pythonhosted.org/packages/ac/35/17c9141c4ae21e9a29a43acdfd848e3e468a810517f862cad07977bf8fe9/google-3.0.0-py2.py3-none-any.whl (45kB)
    100% |████████████████████████████████| 51kB 2.0MB/s 
Requirement already satisfied: beautifulsoup4 in /usr/lib/python3/dist-packages (from google) (4.7.1)
Installing collected packages: google
Successfully installed google-3.0.0
pi@raspberrypi:~/Electron-Cash-4.2.0 $ ./electron-cash
Error: No module named 'google'. Try 'sudo pip install <module-name>'

I don't know how to proceed from here
When I first came to try to run electron-cash, it complained about the absence of dns, but is it dns or pythondns or dnspython ?
The problem is that incorrect instructions are given by the program which result in the user searching for solutions on the web and they don't work (or maybe they do, but they are not what you are suggesting) and perhaps they compound the problem - see https://bitcointalk.org/index.php?topic=5024252.0

I don't understand your remark about using the source-dist rather than the tarball.  There are two options at https://electroncash.org/ for Linux - the App Image or the Source Code - but ".AppImage is for any Linux 64 bit (it's a self-contained app bundle with everything inside)" and I was intending to run electron cash on the 32 bit Raspberry Pi OS.  True, the statement doesn't preclude its use on a 32 bit OS but I thought that was the intended meaning.

I did not have these problems running the latest version of Electrum - it just worked from the source code.

**Comment by richardamullens** [comment]
I don't know why some of my text appears to be struck out above that was not the intention.

**Comment by richardamullens** [comment]
At https://github.com/Electron-Cash/Electron-Cash/releases/tag/4.2.0 it says
Srcdist is provided as .tar.gz and/or .zip files
**Includes Python dependencies inside in packages/**

But when I extract from the tarball there is no packages directory.

I've no idea when packages/ disappeared but it is in Electron Cash-3.3.2/

**Comment by mhsmith** [comment]
> My text above includes angle brackets but these and what lies between them aren't visible.
> I don't know why some of my text appears to be struck out above that was not the intention.

You can disable the GitHub text formatting by surrounding text with marks \`\`\`like this\`\`\`.

**Comment by richardamullens** [comment]
> > My text above includes angle brackets but these and what lies between them aren't visible.
> > I don't know why some of my text appears to be struck out above that was not the intention.
> 
> You can disable the GitHub text formatting by surrounding text with marks ```like this```.

Thank you, let me see using an example from above
```
sudo python3 -m pip uninstall google
Uninstalling google-3.0.0:
Would remove:
/usr/local/bin/google
/usr/local/lib/python3.7/dist-packages/google-3.0.0.dist-info/*
/usr/local/lib/python3.7/dist-packages/googlesearch/*
Proceed (y/n)?
Your response ('') was not one of the expected responses: y, n
Proceed (y/n)? y
Successfully uninstalled google-3.0.0
pi@raspberrypi:/Electron-Cash-4.2.0 $ python3 -m pip install --user google
Looking in indexes: https://pypi.org/simple, https://www.piwheels.org/simple
Collecting google
Downloading
https://files.pythonhosted.org/packages/ac/35/17c9141c4ae21e9a29a43acdfd848e3e468a810517f862cad07977bf8fe9/google-3.0.0-py2.py3-none-any.whl (45kB)
    100% |████████████████████████████████| 51kB 2.0MB/s 
Requirement already satisfied: beautifulsoup4 in /usr/lib/python3/dist-packages (from google) (4.7.1)
Installing collected packages: google
Successfully installed google-3.0.0
pi@raspberrypi:~/Electron-Cash-4.2.0 $ ./electron-cash
Error: No module named 'google'. Try 'sudo pip install <module-name>'
```
Great, thank you :)

**Comment by richardamullens** [comment]
Sorry, I'm slowly getting used to this.  I retired 12 years ago.

**Comment by richardamullens** [comment]
I think that I will close this as I managed to get the source code of 4.2.2 running on the 32bit Raspberry Pi OS (Debian Buster  derived version of 20th August 2020).

To do this I first installed a clean Operating system (my attempts to fix the issue with the missing dns module had damaged the python3 installation).  Next:
`    sudo apt-get install python3-setuptools
     sudo python3 setup.py install
    ./electron-cash
`
This gave me a warning message that PyQt5 version 5.12+ was needed but version 5.11.3 was present - which I ignored
I then chose the last resort option (it was late at night) of 
`sudo apt install libsecp256k1-0`
after which I had a working system.

I think that the following from the Download Notes at [(https://www.reddit.com/r/btc/comments/jx9ba2/electron_cash_422_with_cashfusion_is_now/)] is untrue and misleading
`Srcdist is provided as .tar.gz and/or .zip files
   Includes Python dependencies inside in packages/`

Thanks to all who have helped me by responding to this issue and for continuing to develop the electron cash wallet
